app/controllers/application_controller.rb

```diff
@@ -267,7 +268,28 @@ def invalid_auth_token(_exception)
       user_signed_in: user_signed_in?,
     )
     flash[:error] = t('errors.general')
-    redirect_back fallback_location: new_user_session_url, allow_other_host: false
+    begin
+      redirect_back fallback_location: new_user_session_url, allow_other_host: false
+    rescue
+      analytics.unsafe_redirect_error(
+        controller: controller_info,
+        user_signed_in: user_signed_in?,
+        referer: request.referer,
+      )
+      redirect_to new_user_session_url
+    end
```

Comment:
Could we reuse the method here?
```suggestion
    rescue ActionController::Redirecting::UnsafeRedirectError => err
      unsafe_redirect_error(err)
    end
```

Reply:
Smart, thank you